rego/rules/arm/monitor/key_vault_logging.rego

```diff
@@ -37,20 +37,42 @@ input_type = "arm"
 
 resource_type = "MULTIPLE"
 
-resources = fugue.resources("Microsoft.Insights/diagnosticSettings")
+key_vaults := fugue.resources("Microsoft.KeyVault/vaults")
+diagnostic_settings := fugue.resources("Microsoft.Insights/diagnosticSettings")
 
-is_invalid(resource) {
-	resource.TODO == "TODO" # FIXME
+retention_is_valid(retention) {
+	retention.enabled == true
+	retention.days >= 180
+}
+
+retention_is_valid(retention) {
+	retention.enabled == true
+	retention.days == 0
+}
+
+valid_key_vault_diagnostics := {id: ds |
+	ds := diagnostic_settings[id]
+	contains(lower(ds.scope), "microsoft.keyvault/vaults")
+	log = ds.properties.logs[_]
+	lower(log.category) == "auditevent"
+	log.enabled == true
+	retention_is_valid(log.retentionPolicy)
+}
+
+valid_key_vaults := {id |
+	kv := key_vaults[id]
+	ds := valid_key_vault_diagnostics[_]
+	contains(lower(ds.scope), lower(kv.name))
```

Comment:
I'm concerned this would lead to false matches for things that are prefixes, e.g. `invalidName` and `invalid`.  Maybe a better way is to split `ds.scope` on any whitespace and check if any of the resulting tokens matches the name?  Similar to what we do here: https://github.com/fugue/regula/blob/99ac9082b5a03aa320ead722162e7a8f980ebb1d/rego/lib/fugue/resource_view/arm.rego#L125

Reply:
That's a good point. I think we'll end up with some false failures from that approach, but that's probably better than false passes.

Reply:
Fixed in 213ccbd